compiler/src/dmd/dstruct.d

```diff
@@ -479,6 +479,30 @@ extern (C++) class StructDeclaration : AggregateDeclaration
         return (ispod == ThreeState.yes);
     }
 
+    /***************************************
+     * Determine if struct has copy construction (copy constructor or postblit)
+     * Returns:
+     *     true if struct has copy construction
+     */
+    final bool hasCopyConstruction()
```

Comment:
This function is not necessary. If you reach the gluelayer, semantic has already been performed so you should just check if the struct declaration has `postblit` or `hasCopyCtor` true. There is no need to recursively check if the fields have copy constructors because that's already been taken care of in [buildPostblit](https://github.com/dlang/dmd/blob/master/compiler/src/dmd/clone.d#L1222) or [buildCopyCtor](https://github.com/dlang/dmd/blob/master/compiler/src/dmd/clone.d#L1677) which are called [during the semantic of struct declaration](https://github.com/dlang/dmd/blob/master/compiler/src/dmd/dsymbolsem.d#L4663).